server/apis/v1/handler.go

```diff
@@ -1062,6 +1179,26 @@ func getIsbServices(h *handler, namespace string) (ISBServices, error) {
 	return isbList, nil
 }
 
+// getMonoVertices is a utility used to fetch all the mono vertices in a given namespace
+func getMonoVertices(h *handler, namespace string) (MonoVertices, error) {
+	mvtList, err := h.numaflowClient.MonoVertices(namespace).List(context.Background(), metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+	var resList MonoVertices
+	for _, mvt := range mvtList.Items {
+		status, err := getMonoVertexStatus(&mvt)
+		if err != nil {
+			return nil, err
+		}
+		// NOTE: we only calculate mono vertex lag for get single mono vertex API
+		// to avoid massive gRPC calls
+		resp := NewMonoVertexInfo(status, nil, &mvt)
+		resList = append(resList, resp)
```

Comment:
I do not follow this `NOTE:` comment

Reply:
It was copied from Pipeline lag calculation. I will remove it.